broker-core/src/main/java/io/zeebe/broker/workflow/processor/CatchEventBehavior.java

```diff
@@ -74,33 +86,78 @@ public void subscribeToCatchEvents(
     }
   }
 
-  // CATCH EVENTS
-  private final WorkflowInstanceRecord workflowInstanceRecord = new WorkflowInstanceRecord();
+  public boolean eventOccurred(
+      long elementInstanceKey,
+      DirectBuffer eventHandlerId,
+      DirectBuffer eventPayload,
+      TypedStreamWriter streamWriter) {
 
-  public void triggerCatchEvent(
-      WorkflowInstanceRecord source,
-      DirectBuffer elementId,
-      DirectBuffer payload,
-      TypedStreamWriter writer) {
-    workflowInstanceRecord.wrap(source);
-    workflowInstanceRecord.setPayload(payload);
-    workflowInstanceRecord.setElementId(elementId);
+    final ElementInstanceState elementInstanceState =
+        state.getWorkflowState().getElementInstanceState();
+    final StoredRecord tokenEvent = elementInstanceState.getTokenEvent(elementInstanceKey);
+    final ElementInstance elementInstance = elementInstanceState.getInstance(elementInstanceKey);
+
+    if (tokenEvent != null && tokenEvent.getPurpose() == Purpose.DEFERRED_TOKEN) {
```

Comment:
In which case do we have a token event? It seems this is called from `CorrelateWorkflowInstanceSubscription` and `TriggerTimerProcessor`, neither of which defer tokens before. In which case would I have a token here?

Reply:
There is a deferred token when the event belongs to an intermediate catch event. I can document this :)